precompiles/types/address_test.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/node/testutil/sample"
+)
+
+func Test_GetEVMCallerAddress(t *testing.T) {
+	mockEVM := vm.EVM{
+		TxContext: vm.TxContext{
+			Origin: common.Address{},
+		},
+	}
+
+	mockVMContract := vm.NewContract(
+		contractRef{address: common.Address{}},
+		contractRef{address: common.Address{}},
+		big.NewInt(0),
+		0,
+	)
```

Comment:
I am wondering if it would be helpful to have some negative tests as well, 
something like 
caller, err := GetEVMCallerAddress(&mockEVM, nil)

Reply:
Expanded test cases in 06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf